libraries/lib-effects/EffectOutputTracks.cpp

```diff
@@ -50,7 +50,8 @@ EffectOutputTracks::EffectOutputTracks(
    {
       using namespace BasicUI;
       auto progress = MakeProgress(
-         XO("Pre-processing"), XO("Rendering Time-Stretched Audio"),
+         WaveTrackUtilities::stretchRenderingProgressBarTitle,
+         WaveTrackUtilities::stretchRenderingProgressBarMessage,
```

Comment:
This commit eliminates some duplication.  But it doesn't do enough.

See this in `EffectOutputTracks.cpp`:

```
      auto reportProgress = [&](double progressFraction) {
         const auto overallProgress = (count + progressFraction) / numTracks;
         const auto result = progress->Poll(overallProgress * 1000, 1000);
         if (result != ProgressResult::Success)
            throw UserException {};
      };
```

This lambda needs to be defined only once, and reusable.  In the other two places that use the strings, they repeat part of this lambda, but not the reset of it -- therefore those progress indicators have non-functional Cancel buttons!

The utility should take a reference to `count` and construct a lambda that captures it.  The utility returns a `std::function` and comments that it may throw `UserException`.

Reply:
There is a fourth place -- in AutoDuck.  The identical lambda could be eliminated.

The utility could construct the progress indicator.  Do you know C++14 extended lambda capture?  It lets a lambda hold a `unique_ptr` which C++11 could not do.

```
[
   progress = MakeProgress(...),
  ...
]
```

There is need to vary the progress bar message for AutoDuck but nothing else.  So the utility could take that as another parameter.

Reply:
You would need another parameter too, the denominator of the division.